supertokens_python/recipe/session/interfaces.py

```diff
@@ -589,20 +599,23 @@ def __init__(self, is_valid: bool, reason: Optional[Dict[str, Any]] = None):
 
 
 class SessionClaimValidator(ABC):
-    def __init__(self, id_: str):
+    def __init__(
+        self,
+        id_: str,
+        claim: SessionClaim[_T],  # pyright: ignore[reportInvalidTypeVarUse]
+    ) -> None:
         self.id = id_
-        self.claim: Optional[
-            SessionClaim[Any]
-        ] = None  # Child class must set this if required.
+        self.claim = claim
 
     @abstractmethod
     async def validate(
         self, payload: JSONObject, user_context: Dict[str, Any]
     ) -> ClaimValidationResult:
         pass
 
-    def should_refetch(  # pylint: disable=no-self-use
+    @abstractmethod
+    def should_refetch(
         self, payload: JSONObject, user_context: Dict[str, Any]
     ) -> MaybeAwaitable[bool]:
-        _, __ = payload, user_context
-        return False
+        # TODO: Confirm that MaybeAwaitable actually makes the function async
+        pass
```

Comment:
TODO: Confirm that MaybeAwaitable actually makes the function async
        pass

Reply:
Based on what I've found, `MaybeAwaitable` only affects what type checking tools understand. It doesn't directly make the object asyncio compatible, that happens only when 

But using `MaybeAwaitable` gives us freedom to handle check for things like `isinstance(obj, Awaitable)` in our util function `resolve()`. So it's useful to keep it. There's only one drawback that when it comes to overriding the function in child classes, pyright starts complaining if you use `async` (which actually makes it asynchronous) because of signature differences.

This is a bug in pyright, but not in the python runtime. Infact, some of our tests are using `async def fetch_value` and some are just `def fetch_value`. All of them are passing.

So it's completely fine. We just need to provide `type: ignore` if we are using `async def should_refetch()` in child classes.